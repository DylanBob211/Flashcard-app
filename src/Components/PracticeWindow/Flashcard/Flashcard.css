.flashcard_container {
  display: flex;
  flex-direction: column;
  text-align: center;
}

.flashcard_title {
  font-size: 2rem;
}

.flashcards_imgContainer {
  
}

.flashcard_img {

}