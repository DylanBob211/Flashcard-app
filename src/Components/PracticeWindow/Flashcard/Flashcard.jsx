import React from 'react';
import PropTypes from 'prop-types';

const NO_IMG = require('../../../Assets/imgs/no_img.svg');

const Flashcard = ({ data }) => {
  const { word, url } = data[0];

  const renderImg = () => {
    if (url[0] === 'No Img Available' || url[0] === '') {
      return <img className="wordPreview_img--small" src={NO_IMG} alt="no Word Preview" />;
    }
    return (
      <div className="wordPreview_imgContainer">
        {url.map((itemUrl, ind) => (<img key={`${ind}-${itemUrl}`} className="wordPreview_img--small" src={itemUrl} alt="preview of pictures about this word" />))}
      </div>
    );
  };

  return (
    <>
      <h3 className="wordPreview_title">{ word }</h3>
      {renderImg()}
    </>
  );

};

Flashcard.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({
    word: PropTypes.string,
    url: PropTypes.arrayOf(PropTypes.string),
  })).isRequired,
};

export default Flashcard;
