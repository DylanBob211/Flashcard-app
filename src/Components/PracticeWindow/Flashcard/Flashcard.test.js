import testWrapper from '../../../Helpers/testComponentMountWrapper';
import Flashcard from './Flashcard';

const NO_IMG = require('../../../Assets/imgs/no_img.svg');

describe('Flashcard', () => {
  let mountedFlashcard;
  const props = {};
  const wordTest = 'Hello Word';
  const urlTest = ['urlOne', 'urlTwo', 'urlThree'];

  beforeEach(() => {
    mountedFlashcard = undefined;
  });

  it('matches to snapshot', () => {
    expect(testWrapper(Flashcard, props, true)).toMatchSnapshot();
  });

  describe('always', () => {
    beforeEach(() => {
      mountedFlashcard = testWrapper(Flashcard, props);
    });

    it('renders "(no word)"', () => {
      expect(mountedFlashcard.text()).toEqual('(no word)');
    });

    it('renders an img with src set to NO_IMG', () => {
      const img = mountedFlashcard.find('img');
      expect(img.exists()).toEqual(true);
      expect(img.props().src).toEqual(NO_IMG);
    });

    describe('when `data` is passed', () => {
      beforeEach(() => {
        props.data = {
          word: wordTest,
          url: urlTest,
        };
        mountedFlashcard = testWrapper(Flashcard, props);
      });

      it('renders `word` key of `data` as text', () => {
        expect(mountedFlashcard.text()).toEqual(wordTest);
      });

      it('renders a container with imgs in it', () => {
        const imgContainer = mountedFlashcard.find('.flashcard_imgContainer');
        expect(imgContainer.exists()).toEqual(true);
      });

      it('renders an img for each element in the `url` array of `data`', () => {
        const imgContainer = mountedFlashcard.find('.flashcard_imgContainer');
        expect(imgContainer.children().length).toEqual(3);
      });
    });
  });
});
