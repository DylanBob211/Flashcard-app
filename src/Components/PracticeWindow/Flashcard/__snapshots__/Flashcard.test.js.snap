// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flashcard matches to snapshot 1`] = `
<div
  className="flashcard_container"
>
  <h3
    className="flashcard_title"
  >
    (no word)
  </h3>
  <div
    className="flashcard_imgContainer"
  >
    <img
      alt="no Word Preview"
      className="flashcard_img"
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "no_img.svg",
        }
      }
    />
  </div>
</div>
`;
